### PR TITLE
Add jvm observ lib

### PR DESCRIPTION
**Pull request by v-zhuravlev**
A. Add JVM observability lib  (and process-observ-lib):

1) with the following signal categories:

- JVM memory (heap,nonheap)
- JVM garbage collection
- JVM buffers 
- JVM threads
- JVM classes
- Hikari pools
- Logback stats
- Also, general process stats (uptime,cpu, memory...) (as separate library)

2) with basic JVM alerts
3) with supporting following metric sources:
- OpenTelemetry https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/docs/target-systems/jvm.md transformed to Prometheus via otelcollector exporter(https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusexporter/README.md)
- Prometheus https://prometheus.github.io/client_java/instrumentation/jvm/
- Springboot(micrometer) metrics collected through Prometheus actuator https://docs.micrometer.io/micrometer/reference/implementations/prometheus.html .

B. Convert jvm-mixin to use jvm-observ-lib.
C. Convert spring-boot-mixin to use jvm-observ-lib.

Requires #1268 to be merged first.

image

### CODE REVIEW

**Comment by v-zhuravlev** [comment]
Depends on  https://github.com/grafana/grafonnet/pull/214

File: jvm-observ-lib/dashboards.libsonnet [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        + g.dashboard.withVariables(this.signals.memory.getVariablesMultiChoice())
+        + g.dashboard.withTags(this.config.dashboardTags)
+        + g.dashboard.withUid(this.config.uid + '-jvm-dashboard')
+        + g.dashboard.withPanels(
+          g.util.panel.resolveCollapsedFlagOnRows(
+            g.util.grid.wrapPanels(
+              std.flattenArrays(
+                [
+                  this.grafana.rows.overview,
+                  this.process.grafana.rows.process,
```
Comment (rgeyer) [inline jvm-observ-lib/dashboards.libsonnet:16]: Somehow, this is causing the linting/compilation error where `panels` is not defined. I haven't been able to sort it out tho.

**Review by rgeyer — CHANGES_REQUESTED** [review]
Looks syntactically good, and the structure is good.

There is an error while running the linting, which I've isolated, but can't seem to solve. 🤔

File: jvm-observ-lib/dashboards.libsonnet (hunk 1 above)
Comment (v-zhuravlev) [reply jvm-observ-lib/dashboards.libsonnet:16]: thanks! I'll fix it

Comment (v-zhuravlev) [reply jvm-observ-lib/dashboards.libsonnet]: there was a small bug in commonlib, which I had fixed here.

**Review by rgeyer — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**